POP.nix

```diff
@@ -97,9 +97,11 @@
     precedenceList = computePrecedenceList instantiator meta;
     defaults = lib.foldr mergeInstance bottomInstance (map getDefaults precedenceList);
     __meta__ = meta // {inherit precedenceList;};
-    proto = composeProtos ([(topProto __meta__)] ++ (map getProto precedenceList));
+    metaProto = topProto __meta__;
+    proto = composeProtos ([metaProto] ++ (map getProto precedenceList));
+    self = instantiateProto proto defaults;
   in
-    instantiateProto proto defaults;
+    self // {unpop = unpop self;};
```

Comment:
I'd rather this be done by the `topProto` function, and the field be called `__unpop__` or something instead of `unpop`.